### PR TITLE
CXP-1317: Update Internal/GetCatalogAction

### PR DESCRIPTION
**Pull request by eoliveauakeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Application/Persistence/Catalog/GetCatalogQueryInterface.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+use Akeneo\Catalogs\Domain\Catalog;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface GetCatalogQueryInterface
+{
+    public function execute(string $catalogId): Catalog;
```
Comment (willy-ahva) [inline components/catalogs/back/src/Application/Persistence/Catalog/GetCatalogQueryInterface.php:15]: @throws phpdoc missing, here it's important (it's not always)

File: components/catalogs/back/src/Application/Exception/CatalogNotFoundException.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Infrastructure\Exception;
```
[the remark this answers is not in the document]
Comment (tseho) [reply components/catalogs/back/src/Application/Exception/CatalogNotFoundException.php]: Why ?

[the remark this answers is not in the document]
Comment (RiffVentura) [reply components/catalogs/back/src/Application/Exception/CatalogNotFoundException.php]: or at least to the application level

[the remark this answers is not in the document]
Comment (eoliveauakeneo) [reply components/catalogs/back/src/Application/Exception/CatalogNotFoundException.php]: > or at least to the application level

It's in Akeneo\Catalogs\Application\Exception, the conversation is on an outdated file

[the remark this answers is not in the document]
Comment (willy-ahva) [reply components/catalogs/back/src/Application/Exception/CatalogNotFoundException.php]: The query was in the Domain, it was part of the aggregate too. My thought is that it should be in the aggregate as it belongs to a Catalog. I'm ok to have it in the Application. But I think more and more that we have too many things in Application.

**Review by willy-ahva — APPROVED** [review]

**Review by tseho — APPROVED** [review]

**Comment by tseho** [comment]
Github seems to be outdated on the CI status, it's green: https://app.circleci.com/pipelines/github/akeneo/pim-community-dev/41876/workflows/8bd5be03-155a-4d2b-bc24-cc2b480f9e21
I'm bypassing the requirement to merge.

**Pull request merged**